File: Week 2/Caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    int key, intchar;
    if(argc==2 && atoi(argv[argc-1])!= 0)
    {
        key= atoi(argv[argc-1]);
        string text= get_string("plaintext: ");
        printf("ciphertext: ");
        for(int i=0;text[i] != '\0';i++)
            {
                if(text[i]<= 'Z' && text[i]>= 'A')
                    {
                       text[i]= text[i] + key % 26;
                       if(text[i] > 'Z')
                       {
                           text[i] = 'A' + (text[i]-key) % 26;
                       }
                    }
                if(text[i]<= 'z' && text[i]>= 'a')
                    { 
                       intchar = text[i] + key % 26;
                       if(intchar > 'z' )
                       {
                           text[i] = intchar - 'z' + 'a' - 1;
                       }
                       else
                       {
                           text[i] = text[i] + key % 26;
                       }   
                    }
                printf("%c", text[i]);
            }
        printf("\n");
    }
    else if(argc==1)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    else
    {
        printf("Usage: ./caesar key\n");
    }
    return 0;
}

File: Week 6/caesar.py
from sys import exit
from sys import argv
from cs50 import get_string

if len(argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)

k = int(argv[1])
text = get_string("plaintext: ")

print("ciphertext: ", end="")
for char in text:
    asciiValue = ord(char)
    if asciiValue >= 65 and asciiValue <= 90:
        newAsciiValue = asciiValue + k % 26
        if (newAsciiValue > 90):
            newAsciiValue -= 26 
        print(chr(newAsciiValue), end="")
    elif asciiValue >= 97 and asciiValue <= 122:
        newAsciiValue = asciiValue + k % 26
        if (newAsciiValue > 122):
            newAsciiValue -= 26 
        print(chr(newAsciiValue), end="")
    else:
        print(char, end="")
print()
exit(0)
